static/js/popovers.js

```diff
@@ -236,19 +236,25 @@ exports.show_user_profile = function (user) {
         profile_field.is_link = field_type === field_types.URL.id;
         profile_field.type = field_type;
         if (field_value) {
-            if (field_type === field_types.DATE.id) {
-                profile_field.value = moment(field_value).format(localFormat);
-            } else if (field_type === field_types.USER.id) {
-                profile_field.id = field.id;
-                profile_field.is_user_field = true;
-                profile_field.value = field_value;
-            } else if (field_type === field_types.CHOICE.id) {
-                var field_choice_dict = JSON.parse(field.field_data);
-                profile_field.value = field_choice_dict[field_value].text;
-            } else {
-                profile_field.value = field_value;
+            if (field_value.value) {
```

Comment:
It'd generally better to use the early-return pattern for this sort of thing, rather than nesting lots of layers of if statements.  Might want to convert the `if (field_value)` thing as well.

Reply:
Sounds good. Will do. Also, how about converting the nested if-else block into a switch statement? This is JavaScript after all.

Reply:
yeah switch would be reasonable.  Do that refactor in a prep commit though.

Reply:
(replied in PM)

Reply:
@hackerkid explained it to me so I'll split up my commits accordingly.